src/AccessToken.php

```diff
@@ -360,14 +359,28 @@ private function retrieveCertsFromLocation($url, array $options = [])
                 ));
             }
 
-            return json_decode((string) file_get_contents($url), true);
+            return [
+                json_decode((string) file_get_contents($url), true),
+                $expireTime
+            ];
         }
 
         $httpHandler = $this->httpHandler;
         $response = $httpHandler(new Request('GET', $url), $options);
 
         if ($response->getStatusCode() == 200) {
-            return json_decode((string) $response->getBody(), true);
+            if ($cacheControl = $response->getHeaderLine('Cache-Control')) {
+                array_map(function ($value) use (&$expireTime) {
+                    list($key, $value) = explode('=', $value) + [null, null];
+                    if ($key == 'max-age') {
+                        $expireTime = '+' . $value . ' seconds';
```

Comment:
this seems to work, but the PHP spec seems to say only sec or second is valid, not seconds? https://www.php.net/manual/en/datetime.formats.php#datetime.formats.relative

Reply:
I believe that is referring to "second" in the sense of "first, second, third...". The keyword "seconds" is definitely valid.

Reply:
Here:

```
(( 'msec' | 'millisecond' | 'µsec' | 'microsecond' | 'usec' | 'sec' | 'second' | 'min' | 'minute' | 'hour' | 'day' | 'fortnight' | 'forthnight' | 'month' | 'year') 's'?)
```
You can see there's the optional `s` at the end, which confirms `seconds` is valid.